src/drag-drop-context.tsx

```diff
@@ -227,18 +228,19 @@ const DragDropProvider: Component<DragDropContextProps> = (passedProps) => {
       }
     });
   };
-  const activeDraggable = (): Draggable | null => {
+
+  const activeDraggable = createMemo((): Draggable | null => {
```

Comment:
I wonder if `untrack` would work better here. It could prevent doing the work at all by ignoring changes to the draggable state object itself and only reacting to the active draggable change.

Reply:
@jpdutoit Actually, I'm not able to reproduce a case where the `activeDraggable` in an effect causes it to fire incorrectly on object change rather than active change. I'm going to revert this until I have a reproducible case - do you have one that triggers this behaviour you can share?

Reply:
I'm not sure when I can isolate a reproducible case.

But, the problem in essence is that an expression like `state.droppables[state.previous.droppable]` tracks both `state.droppables` and the specific droppable.

One specific chain of events where this happens:

1. OnDragEnd triggered correctly via onPointerUp after a drop on droppable with id _A_
2. My code moves an object from container 1 to container 2
3. Old element is unmounted and a droppable inside it with id _B_ is removed
4. Solid-dnd schedules cleanup for the removed droppable _B_
5. Element is recreated in container 2 and droppable inside it with id _C_ is created
6. After addDroppable in step 5, state.droppables registers as changed, leading to spurious call to onDragEnd
7. Cleanup scheduled in step 4 runs, state.droppables registers as changed, leading to another spurious call to onDragEnd

Now, droppable B and C has nothing to do with the drop but it is their creation/destruction that causes the extra calls. The same thing applies if you would create/remove unrelated draggables.

This can also happen any arbitrary time later. For example if you have a button that adds a new element containing draggables/droppables somewhere, it will re-trigger onDragEnd because all effects dependent on state.draggables and/or state.droppables will re-run.

Reply:
Ah, so like [this ](https://playground.solidjs.com/?hash=-81141575&version=1.3.13 )where changing keys in the state object triggers a signal. For some reason I thought Solid didn't notify in that case, but it seems it does.

So I think `untrack` is the right thing to do here rather than run redundant evaluations. I'll update to that. 

Thanks!